hail/src/main/scala/is/hail/io/fs/AzureStorageFS.scala

```diff
@@ -0,0 +1,320 @@
+        val blobFileName = s"hail-az://$account/$container/${blobItem.getName}"
+        this.getBlobClient(blobFileName).delete()
+      })
+
+      if (fileStatus.isFile) blobClient.delete()
+    }
+    else {
+        blobClient.delete()
+    }
+  }
+
+  def listStatus(filename: String): Array[FileStatus] = {
+    val (account, container, path) = getAccountContainerPath(filename)
+
+    val blobContainerClient: BlobContainerClient = getContainerClient(filename)
+    val statList: ArrayBuffer[FileStatus] = ArrayBuffer()
+
+    val prefix = dropTrailingSlash(path) + "/"
+    // collect all children of this directory (blobs and subdirectories)
+    val prefixMatches = blobContainerClient.listBlobsByHierarchy(prefix)
+
+    prefixMatches.forEach(blobItem => {
+      val blobFileName = s"hail-az://$account/$container/${blobItem.getName}"
+      statList += fileStatus(blobFileName)
+    })
+    statList.toArray
+  }
+
+  def glob(filename: String): Array[FileStatus] = {
+    var (account, container, path) = getAccountContainerPath(filename)
+    path = dropTrailingSlash(path)
+
+    val components =
+      if (path == "")
+        Array.empty[String]
+      else
+        path.split("/")
+
+    val javaFS = FileSystems.getDefault
+
+    val ab = new mutable.ArrayBuffer[FileStatus]()
+    def f(prefix: String, fs: FileStatus, i: Int): Unit = {
+      assert(!prefix.endsWith("/"), prefix)
+
+      if (i == components.length) {
+        var t = fs
+        if (t == null) {
+          try {
+            t = fileStatus(prefix)
+          } catch {
+            case _: FileNotFoundException =>
+          }
+        }
+        if (t != null)
+          ab += t
+      }
+
+      if (i < components.length) {
+        val c = components(i)
+        if (containsWildcard(c)) {
+          val m = javaFS.getPathMatcher(s"glob:$c")
+          for (cfs <- listStatus(prefix)) {
+            val p = dropTrailingSlash(cfs.getPath)
+            val d = p.drop(prefix.length + 1)
+            if (m.matches(javaFS.getPath(d))) {
+              f(p, cfs, i + 1)
+            }
+          }
+        } else
+          f(s"$prefix/$c", null, i + 1)
+      }
+    }
+
+    f(s"hail-az://$account/$container", null, 0)
+    ab.toArray
+  }
+
+  def fileStatus(filename: String): FileStatus = {
+    val (account, container, path) = getAccountContainerPath(filename)
+
+    if (path == "") {
+      return new BlobStorageFileStatus(s"hail-az://$account/$container", null, 0, true)
+    }
+
+    val blobClient: BlobClient = getBlobClient(filename)
+    val blobContainerClient: BlobContainerClient = getContainerClient(filename)
+
+    val prefix = dropTrailingSlash(path) + "/"
+    val options: ListBlobsOptions = new ListBlobsOptions().setPrefix(prefix)
+    val prefixMatches = blobContainerClient.listBlobs(options, null)
+    val isDir = prefixMatches.iterator().hasNext
+
+    if (!isDir && !blobClient.exists()) throw new FileNotFoundException("file isn't found")
```

Comment:
nit: can you add `filename` to the error message here and capitalize the first word?